### PR TITLE
Fixed various issues

### PR DESCRIPTION
**Pull request by StrangeZak**

### CODE REVIEW

**Comment by StrangeZak** [comment]
Check this out and see if it woks on your build @bvisness

**Comment by bvisness** [comment]
If you merge #38 we can have Travis run it! 😄

**Comment by bvisness** [comment]
(I am checking it out and running it locally in the meantime.)

File: HandmadeMath.h [hunk 1]
```diff
@@ -334,20 +338,22 @@ typedef hmm_mat4 hmm_m4;
...
 HMMDEF float HMM_Power(float Base, int Exponent);
 HMMDEF float HMM_Clamp(float Min, float Value, float Max);
 
 HMMDEF hmm_vec3 HMM_Normalize(hmm_vec3 A);
 HMMDEF hmm_vec3 HMM_Cross(hmm_vec3 VecOne, hmm_vec3 VecTwo);
-HMMDEF float HMM_Dot(hmm_vec3 VecOne, hmm_vec3 VecTwo);
+
+HMMDEF float HMM_DotVec2(hmm_vec3 VecOne, hmm_vec3 VecTwo);
+HMMDEF float HMM_DotVec3(hmm_vec3 VecOne, hmm_vec3 VecTwo);
+HMMDEF float HMM_DotVec4(hmm_vec3 VecOne, hmm_vec3 VecTwo);
```
Comment (bvisness) [inline HandmadeMath.h]: These function signatures all use `hmm_vec3`.

File: HandmadeMath.h [hunk 2]
```diff
@@ -122,7 +122,15 @@
...
-
+      0.6
+          (*) Added Unit testing
+          (*) Made HMM_Power faster
+          (*) Fixed possible efficiency problem with HMM_Normalize 
+          (*) RENAMED HMM_LengthSquareRoot to HMM_LengthSquared
+          (*) RENAMED HMM_RSqrtF to HMM_RSquareRootF
+          (*) RENAMED HMM_SqrtF to HMM_SquareRootF
+          (*) REMOVED Inner function (user should use Dot now)
+          (*) REMOVED HMM_FastInverseSquareRoot function declaration
```
Comment (bvisness) [inline HandmadeMath.h]: `HMM_FastInverseSquareRoot` is still defined in the headers.

File: HandmadeMath.h (hunk 1 above)
Comment (bvisness) [inline HandmadeMath.h]: Also, let's add C++-style overloaded versions of these called just `HMM_Dot`.

**Comment by bvisness** [comment]
I am not a contributor to this repo, so rather than do some complicated thing with branches and merging, here is a Gist with some changes that should be made to the tests: https://gist.github.com/bvisness/88a9b8b10095e600476bf3c80cdbaf85

**Comment by StrangeZak** [comment]
@bvisness Oh my. Here ill make you a contributor now

**Comment by StrangeZak** [comment]
@bvisness You shouldve gotten a email or something now that i added you

**Comment by StrangeZak** [comment]
@bvisness Wanna do a final check on this before i merge this into master, i think were good but i would like someone else to check on it.

**Comment by bvisness** [comment]
I found a quick problem with the tests that I'm fixing. Give me just a bit.

**Comment by StrangeZak** [comment]
Can Travis run a MSVC build also?

**Comment by bvisness** [comment]
Sadly, no. :( It only supports Make with clang and gcc at the moment. Still, that should be enough to verify that everything works in general (although warnings may vary.)

**Comment by StrangeZak** [comment]
Makes sense, everything good to merge in now?

File: test/hmm_test.cpp [hunk 3]
```diff
@@ -47,28 +42,12 @@ TEST(ScalarMath, ToRadians)
...
-    EXPECT_FLOAT_EQ(HMM_Power(2.0f, 0), 1.0f);
-    EXPECT_FLOAT_EQ(HMM_Power(2.0f, 4), 16.0f);
-    EXPECT_FLOAT_EQ(HMM_Power(2.0f, -2), 0.25f);
-}
-
-TEST(ScalarMath, Clamp)
-{
-    EXPECT_FLOAT_EQ(HMM_Clamp(-2.0f, 0.0f, 2.0f), 0.0f);
-    EXPECT_FLOAT_EQ(HMM_Clamp(-2.0f, -3.0f, 2.0f), -2.0f);
-    EXPECT_FLOAT_EQ(HMM_Clamp(-2.0f, 3.0f, 2.0f), 2.0f);
```
Comment (bvisness) [inline test/hmm_test.cpp]: Why did you remove the test for `HMM_Clamp`?

**Comment by bvisness** [comment]
Besides the test for `HMM_Clamp` being deleted, everything looks good to merge! (And it works fine in my project, after renaming some functions.)

**Comment by StrangeZak** [comment]
Merged
